### PR TITLE
feat: The Odin Project function basics

### PR DESCRIPTION
**Pull request by Sembauke**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: curriculum/challenges/_meta/top-learn-function-basics/meta.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "TOP Learn function basics",
+  "isUpcomingChange": true,
```
Comment (moT01) [inline curriculum/challenges/_meta/top-learn-function-basics/meta.json:3]: Do we want this to be visible when it gets merged or keep it hidden for now?

Looks like a few of the other blocks are still hidden as well. Not sure if that's intentional or not.

**Review by moT01 — APPROVED** [review]
Hope you don't mind @Sembauke, I pushed a few formatting adjustments and some other things. Take a look at the commit to see. Let me know if we don't want any of it or you have any questions about any of them.

Don't forget to complete the checklist in the PR description 😉 

LGTM 🎉

**Review by naomi-lgbt — APPROVED** [review]

File: curriculum/challenges/_meta/top-learn-function-basics/meta.json (hunk 1 above)
Comment (naomi-lgbt) [reply curriculum/challenges/_meta/top-learn-function-basics/meta.json:3]: I think we can make a separate PR to sweep through those and audit as needed.

Comment (Sembauke) [reply curriculum/challenges/_meta/top-learn-function-basics/meta.json:3]: Yep that is intentional, I want to release the javascript part all at once.

**Pull request merged**